proposals/accepted/047-standard-library-reform.rst

```diff
@@ -0,0 +1,559 @@
+
+The much lower CPP count for Web Assembly reflects that fact that the reference `WASI libc`_ itself tries to emulate POSIX the best it can.
+But this just means the same infelicities are there, just less directly observable.
+For example, it incorporates the techniques of `libpreopen`_ to simulate ambient authority such as opening arbitrary files by absolute path.
+But best-effort techniques like this only if one is lucky; they are a great way for adapting *existing* applications but a *poor* way for writing new greenfield ones.
+
+.. _`WASI libc`: https://github.com/WebAssembly/wasi-libc
+.. _`libpreopen`: https://github.com/musec/libpreopen
+
+Solution criteria
+^^^^^^^^^^^^^^^^^
+
+Projects should be able to depend on libraries that just expose functionality that is known to work on the platform(s) they run on.
+The plural, "platforms" is key.
+Projects that wish to support some subset of Unix, Windows, Web, and WASI must be able to depend on libraries that only offer the *intersection* of what works on each of those, i.e. what works on all of them.
+We will thus need more than one standard library.
+
+Platform-specific functionality should be exposed in ways that make sense in Haskell, not C.
+Traditional libc idioms and "lowest common denominator" practice should be skipped when it does not make sense in a Haskell context.
+It should be possible to use WASM and WASI without any "libc".
+
+**Problem 4**: Breaking changes have to coupled, not staggered, with GHC versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Wishful thinking would have it that we can just *stop* doing breaking changes, forever.
+But requirements change, and mistakes are made.
+Issues will arise in the standard library and we will wish to fix them, because whatever the cost is to existing programs (which we can still attempt to mitigate) is outweighed by the benefit to future programs.
+
+However, if the standard library version is tied to GHC version, we have no choice but to do the breaking change coupled with a compiler version.
+Gabriella Gonzalez laid out the case in `Release early and often <https://www.haskellforall.com/2019/05/release-early-and-often.html>`_ on why coupling changes, especially breaking changes, together is bad, and I will cite that rather than restate the argument.
+For those reasons we shouldn't do that here with the standard library and GHC.
+
+Solution criteria
+^^^^^^^^^^^^^^^^^
+
+Changes in the standard library in the compiler should always be staggered.
+It should be possible to upgrade the compiler with only a minor version change or less in the standard library.
+It should likewise be possible to upgrade a major version change in the standard library without breaking a compiler.
+
+**Problem 5**: Popular and uncontroversial machinery like ``Text`` not available from the standard library
```

Comment:
Aren't they? What is the standard library? Is it the stuff in scope by default after `cabal init`? If you just mean `base` then say that.